provider/maas/environ.go

```diff
@@ -2246,15 +2246,39 @@ func (env *maasEnviron) allocateContainerAddresses2(hostInstanceID instance.Id,
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	createDeviceArgs := gomaasapi.CreateMachineDeviceArgs{
-		Hostname:      deviceName,
-		MACAddress:    primaryMACAddress,
-		Subnet:        subnet, // can be nil
-		InterfaceName: primaryNICName,
+	// Check to see if we've already tried to allocate information for this device:
+	devicesArgs := gomaasapi.DevicesArgs{
+		Hostname: []string{deviceName},
 	}
-	device, err := machine.CreateDevice(createDeviceArgs)
+	var device gomaasapi.Device
+	maybeDevices, err := machine.Devices(devicesArgs)
 	if err != nil {
-		return nil, errors.Trace(err)
+		logger.Warningf("error while trying to lookup %q: %v", deviceName, err)
+	} else {
+		if len(maybeDevices) == 1 {
+			logger.Debugf("found MAAS device for container %q, using existing device", deviceName)
+			device = maybeDevices[0]
+		} else if len(maybeDevices) > 1 {
+			logger.Warningf("found too many MAAS devices (%d) for container %q", len(maybeDevices), deviceName)
```

Comment:
logger.Warningf("found more than 1 MAAS devices (%d) for container %q", len(maybeDevices), deviceName)

better?